test/e2e/framework/providers/vsphere/vsphere.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+func init() {
+	framework.RegisterProvider("vsphere", newProvider)
+}
+
+func newProvider() (framework.ProviderInterface, error) {
+	return &framework.NullProvider{}, nil
```

Comment:
this is essentially a `NullProvider`, what is the plan for `vsphere` provider after this change?
otherwise, you could use either `local` or `skeleton` provider in your configure (both are same as `NullProvider`).

Reply:
i noticed a similar change from openstack, https://github.com/kubernetes/kubernetes/pull/74370, if this is something we are trying to achieve, i am good with the change (to support vsphere provider on some e2e storage tests).

cc @BenTheElder

Reply:
`skeleton` should be the preferred null provider going forward 👍 
we did allow openstack in due to the storage tests though. cc @dims

Reply:
This change is indeed to support e2e vsphere storage tests.
https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/vsphere

Multiple tests in this folder are configured to skip the test if the provider specified isn't vsphere.
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/vsphere/vsphere_zone_support.go

`	BeforeEach(func() {
		framework.SkipUnlessProviderIs("vsphere")
		Bootstrap(f)`